### PR TITLE
[ROCm] fix dtype

### PR DESCRIPTION
**Pull request by yiakwy-xpu-ml-framework-team**
## Motivation

fix fp8 dtype problem introduced in https://github.com/sgl-project/sglang/pull/4485

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

**Comment by yiakwy-xpu-ml-framework-team** [comment]
cc @HaiShaw

**Comment by hebiao064** [comment]
Thanks a lot!

File: python/sglang/srt/layers/quantization/w8a8_fp8.py [hunk 1]
```diff
@@ -108,10 +108,15 @@ def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
                     layer.weight, layer.weight.shape[-1]
                 )
                 weight_scale = weight_scale.t().contiguous()
+                if _is_hip:
```
Comment (hebiao064) [inline python/sglang/srt/layers/quantization/w8a8_fp8.py:111]: does AMD also supported Cutlass? I thought it's not...

Comment (yiakwy-xpu-ml-framework-team) [reply python/sglang/srt/layers/quantization/w8a8_fp8.py:111]: CK_tile (cutlass in AMD I personally believe) is currently recommended to program complex kernel fusion.

It contains tile window - ddr window transformation, pipeliner and partitioner. Very useful. You can check it out. D

And in the recently ASM kernels, they added many asm level optimizations.
